src/vault/deposit.ts

```diff
@@ -0,0 +1,42 @@
+// ============ External Imports ============
+import { BigNumber, ContractReceipt, ethers } from "ethers";
+
+// ============ Internal Imports ============
+import vaultAbi from "../../abi/Vault.json";
+
+export abstract class Vault {
+    static async deposit(
```

Comment:
We need to think of how we can input the right amount of tokens. We need an SDK functionality that fetches amount2 based on amount1 and vice versa.